packages/general/src/storage/StorageContext.ts

```diff
@@ -107,15 +67,74 @@ export class StorageContext<S extends Storage = any> implements StorageContextFa
             }
         });
         if (promises.length > 0) {
-            return MatterAggregateError.allSettled(promises, "Error while clearing storage").then(
-                () => {},
-            ) as StorageOperationResult<S>;
+            return MatterAggregateError.allSettled(promises, "Error while clearing storage").then(() => {});
         }
-        return undefined as StorageOperationResult<S>;
+        return undefined;
     }
 
     /** Clears all keys in this context and all created sub-contexts. */
     clearAll() {
-        return this.storage.clearAll(this.thisContexts) as StorageOperationResult<S>;
+        return this.storage.clearAll(this.thisContexts);
+    }
+}
+
+export class StorageContext extends BaseStorageContext {
+    get<T extends SupportedStorageTypes>(key: string, defaultValue?: T): MaybePromise<T> {
+        const value = this.storage.get(this.thisContexts, key);
+        if (value !== undefined) {
+            if (MaybePromise.is(value)) {
+                return value.then(v => {
+                    if (v !== undefined) return v;
+                    if (defaultValue === undefined) {
+                        throw new StorageError(
+                            `No value found for key ${key} in context ${this.thisContexts} and no default value specified!`,
+                        );
+                    }
+                    return defaultValue as T;
+                }) as MaybePromise<T>;
+            }
+            return value as T;
+        }
+        if (defaultValue === undefined) {
+            throw new StorageError(
+                `No value found for key ${key} in context ${this.thisContexts} and no default value specified!`,
+            );
+        }
+        return defaultValue;
+    }
+
+    set(key: string, value: SupportedStorageTypes): MaybePromise<void>;
+    set(values: Record<string, SupportedStorageTypes>): MaybePromise<void>;
+    set(keyOrValues: string | Record<string, SupportedStorageTypes>, value?: SupportedStorageTypes) {
+        if (typeof keyOrValues === "string") {
+            return this.storage.set(this.thisContexts, keyOrValues, value);
+        }
+        return this.storage.set(this.thisContexts, keyOrValues);
+    }
+
+    values() {
+        return this.storage.values(this.thisContexts);
+    }
+}
+
+export class BlobStorageContext extends BaseStorageContext {
```

Comment:
It's weird having BlobStorageContext not be a StorageContext.

Also, I may have just missed it, but what's the purpose of having BlobStorageContext and StorageContext be separate?  Vs. just putting the blob methods onto StorageContext, since they all proxy through to the same underlying Storage anyway?

Reply:
I guess I'd take one of two approaches:

1. Implement blobs as a type of value.  Add minimal extension for `Blob` type values and treat the lack of string<->blob coercion as a (possibly temporary) inconvenience that results in a runtime error

2. Implement blob storage as a completely separate API based on standard `FilesystemDirectoryHandle`.  At some point in the future, replace the node-specific `StorageContext` implementation with one that builds on this

I'm thinking it'd be nice to avoid making our current k/v API more confusing...  I already find it kind of clunky

Reply:
Check newest variant please